ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js

```diff
@@ -1,40 +1,65 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import PropTypes from "prop-types";
 
 import "./RepositoryForm.scss";
+import actions from "app/settings/actions";
+import selectors from "app/settings/selectors";
 import Button from "app/base/components/Button";
 import Col from "app/base/components/Col";
 import Card from "app/base/components/Card";
+import Loader from "app/base/components/Loader";
 import Row from "app/base/components/Row";
 
 export const RepositoryForm = ({ title }) => {
+  const generalLoaded = useSelector(selectors.general.loaded);
+  const reposLoaded = useSelector(selectors.repositories.loaded);
+
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (!reposLoaded) {
+      dispatch(actions.repositories.fetch());
+    }
+    if (!generalLoaded) {
```

Comment:
Currently, these actions all get dispatched twice.

There is a more sinister issue though - given _any_ general fetch action will set `state.general.loaded`, if another component dispatched a general fetch action (not even one of the ones here!), these three actions would never be dispatched.

I think the problem here is that these items in state are all sharing the same `loaded` property, rather than having their own.

Reply:
Hmm I guess I can create reducers for each piece of general data, each with their own loading and loaded fields, then the payload from success could live in a data field, like:
``` js
{
  osInfo: {
    loaded: false,
    loading: false,
    data: {}
  },
  knownArchitectures: {
    loading: false,
    loaded: false,
    data: []
  }
...